### PR TITLE
DHFPROD-9752: Convert datahub 5 directory to JavaScript Module (mjs)

### PR DESCRIPTION
**Pull request by prolonMarkLogic**
#### Checklist: 
```diff
- Note: do not change the below
```

- ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [n/a] Added Tests
- [n/a] Ran cypress tests on firefox locally
  

- ##### Reviewer:

- [ ] Reviewed Tests
- [n/a] Added to Release Wiki

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mastering/common.mjs [hunk 1]
```diff
@@ -18,7 +18,7 @@ function retrieveInterceptorFunction(interceptorObj, interceptorType) {
     return interceptorFunction;
 }
 
-function applyInterceptors(interceptorType, accumulated, interceptors, ...additionalArguments) {
+export function applyInterceptors(interceptorType, accumulated, interceptors, ...additionalArguments) {
```
Comment (aniamy) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mastering/common.mjs:21]: Shouldn't we keep the same format and do the export at the bottom with all the functions?

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/datahub.mjs [hunk 2]
```diff
@@ -15,12 +15,12 @@
...
-const consts = require("/data-hub/5/impl/consts.sjs");
+import consts from "/data-hub/5/impl/consts.mjs";
 
 import Flow from "/data-hub/5/impl/flow.mjs";
-const Perf = require("/data-hub/5/impl/perf.sjs");
-const Debug = require("/data-hub/5/impl/debug.sjs");
-const defaultConfig = require("/com.marklogic.hub/config.sjs");
+import Perf from "/data-hub/5/impl/perf.mjs";
+import Debug from "/data-hub/5/impl/debug.mjs";
+import defaultConfig from "/com.marklogic.hub/config.mjs";
```
Comment (aniamy) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/datahub.mjs:23]: This one remains as sjs

**Comment by aniamy** [comment]
@ryanjdew @prolonMarkLogic What are your thoughts on this? import config from "/com.marklogic.hub/config.mjs";
config.sjs is just variables. Do you guys think we should also convert it to mjs?

**Review by aniamy — APPROVED** [review]

**Review by ryanjdew — APPROVED** [review]

**Comment by ryanjdew** [comment]
@aniamy Yes. We should convert config.sjs to mjs as well.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mastering/matching/matchable.mjs [hunk 3]
```diff
@@ -340,7 +340,7 @@ class Matchable {
 
 const cachedPropertyValues = {};
 
-class MatchRulesetDefinition {
+export class MatchRulesetDefinition {
```
Comment (aniamy) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mastering/matching/matchable.mjs:343]: This class is also exported at the bottom. I suggest we export it only once.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/provenance/provenanceWriteQueue.mjs [hunk 4]
```diff
@@ -35,7 +35,7 @@ const provenanceNamespaces = {
  * conflicting update errors and to facilitate incremental changes such as adding new PROV relationships.
  * Trace logging is used to record this so that a user can have visibility into when this happens.
  */
-class ProvenanceWriteQueue {
+export class ProvenanceWriteQueue {
```
Comment (aniamy) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/provenance/provenanceWriteQueue.mjs:38]: Same here about the export. One export should be enough

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/transforms/mlcp-flow-transform.mjs [hunk 5]
```diff
@@ -32,7 +32,7 @@ for (let requestField of xdmp.getRequestFieldNames()) {
 const visitedURIs = [];
 const urisToContent = {};
 
-function transform(content, context = {}) {
+export function transform(content, context = {}) {
```
Comment (aniamy) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/transforms/mlcp-flow-transform.mjs:35]: Same here about the export

**Review by ryanjdew — APPROVED** [review]

**Review by aniamy — APPROVED** [review]

**Pull request merged**